custom/enikshay/management/commands/soft_delete_cases_with_no_person.py

```diff
@@ -46,17 +47,24 @@ def get_case_ids(domain, case_type):
         return CaseAccessors(domain).get_case_ids_in_domain(type=case_type)
 
     @staticmethod
-    def should_delete(domain, case_id):
-        try:
-            return not get_person_case(domain, case_id)
-        except ENikshayCaseNotFound:
-            return True
+    def get_case_ids_to_delete(domain, case_ids):
+        for case_id in with_progress_bar(case_ids):
+            if _is_case_missing_person(domain, case_id):
+                yield case_id
 
     @staticmethod
-    def delete_case(case_id, commit, deletion_id, domain, logger, case_type):
-        _log_case_to_delete(case_id, deletion_id, domain, logger, case_type)
+    def delete_cases(case_ids, commit, deletion_id, domain, logger, case_type):
+        for case_id in case_ids:
+            _log_case_to_delete(case_id, deletion_id, domain, logger, case_type)
         if commit:
-            CaseAccessors(domain).soft_delete_cases([case_id], deletion_id=deletion_id)
+            CaseAccessors(domain).soft_delete_cases([case_ids], deletion_id=deletion_id)
```

Comment:
this doesn't need to be a static list anymore: `[case_ids]` -> `case_ids`

Reply:
thanks, fixed